ofrak_core/src/ofrak/core/android_sparse.py

```diff
@@ -0,0 +1,195 @@
+    """
+
+    async def get_file(self) -> Resource:
+        """
+        Get the unpacked raw image resource.
+
+        :return: The raw (non-sparse) image resource
+        """
+        return await self.resource.get_only_child()
+
+
+class AndroidSparseImageUnpacker(Unpacker[None]):
+    """
+    Unpack (convert) an Android sparse image to raw image format using simg2img.
+    """
+
+    id = b"AndroidSparseImageUnpacker"
+    targets = (AndroidSparseImage,)
+    children = (GenericBinary,)
+    external_dependencies = (SIMG2IMG,)
+
+    async def unpack(self, resource: Resource, config=None):
+        """
+        Unpack the Android sparse image by converting it to raw format.
+
+        :param resource: The sparse image resource to unpack
+        :param config: Optional unpacker configuration (unused)
+        """
+        async with resource.temp_to_disk() as sparse_path:
+            with tempfile.NamedTemporaryFile(
+                suffix=".img", mode="rb", delete_on_close=False
+            ) as raw_file:
+                raw_file.close()
+
+                cmd = [
+                    "simg2img",
+                    sparse_path,
+                    raw_file.name,
+                ]
+                proc = await asyncio.create_subprocess_exec(
+                    *cmd,
+                    stdout=asyncio.subprocess.PIPE,
+                    stderr=asyncio.subprocess.PIPE,
+                )
+                stdout, stderr = await proc.communicate()
+
+                if proc.returncode:
+                    raise CalledProcessError(returncode=proc.returncode, cmd=cmd, stderr=stderr)
+
+                with open(raw_file.name, "rb") as raw_fh:
+                    raw_data = raw_fh.read()
+
+                await resource.create_child(tags=(GenericBinary,), data=raw_data)
+
+
+class AndroidSparseImagePacker(Packer[None]):
+    """
+    Pack (convert) a raw image back into Android sparse image format using img2simg.
+    """
+
+    id = b"AndroidSparseImagePacker"
+    targets = (AndroidSparseImage,)
+    external_dependencies = (IMG2SIMG,)
+
+    async def pack(self, resource: Resource, config=None):
+        """
+        Pack the raw image back into sparse format.
+
+        :param resource: The sparse image resource to pack
+        :param config: Optional packer configuration (unused)
+        """
+        sparse_view = await resource.view_as(AndroidSparseImage)
+        raw_child_r = await sparse_view.get_file()
+        raw_data = await raw_child_r.get_data()
+
+        with tempfile.NamedTemporaryFile(
+            suffix=".img", mode="wb", delete_on_close=False
+        ) as raw_file:
+            raw_file.write(raw_data)
+            raw_file.close()
+
+            with tempfile.NamedTemporaryFile(
+                suffix=".simg", mode="rb", delete_on_close=False
+            ) as sparse_file:
+                sparse_file.close()
+
+                # Default block size is 4096 bytes (standard for Android)
```

Comment:
@rbs-afflitto, seems like this is a future todo -- detect block size while unpacking, so we can repack to the same block size. Let's not block this MR -- can you make an issue for this though so we can track it?